generators/server/templates/src/main/java/package/security/oauth2/OAuth2AuthenticationService.java.ejs

```diff
@@ -88,9 +90,9 @@ public class OAuth2AuthenticationService {
                 log.debug("successfully authenticated user {}", params.get("username"));
             }
             return ResponseEntity.ok(accessToken);
-        } catch (Exception ex) {
+        } catch (HttpClientErrorException ex) {
             log.error("failed to get OAuth2 tokens from UAA", ex);
-            throw ex;
+            throw new InvalidPasswordException();
```

Comment:
dont we need to pass the ex for stacktrace?

Reply:
The printed stack trace looks mostly the same, main difference is at the end where it returns the correct exception.  

[Logs from before change](https://gist.githubusercontent.com/ruddell/f5a8c406307bf92b231a6a5a93f1c04f/raw/dcd74aa83810534b8d8283060535a8191cc5a53a/gistfile1.txt) 
[Logs from after change](https://gist.githubusercontent.com/ruddell/b6c38a9b84958e5e351563cc72ede523/raw/b4309503930945911eb1f4dd2d071f7abf8d9db1/After%2520Changes%2520StackTrace%2520PR%25208108)